OpenRA.Game/Graphics/WorldRenderer.cs

```diff
@@ -220,34 +247,44 @@ public void Draw()
 				foreach (var r in g)
 					r.Render(this);
 
+			Game.Renderer.Flush();
+
+			for (var i = 0; i < preparedAnnotationRenderables.Count; i++)
+				preparedAnnotationRenderables[i].Render(this);
+
 			if (debugVis.Value != null && debugVis.Value.RenderGeometry)
 			{
 				for (var i = 0; i < preparedRenderables.Count; i++)
 					preparedRenderables[i].RenderDebugGeometry(this);
 
-				foreach (var g in groupedOverlayRenderables)
-					foreach (var r in g)
-						r.RenderDebugGeometry(this);
+				for (var i = 0; i < preparedOverlayRenderables.Count; i++)
+					preparedOverlayRenderables[i].RenderDebugGeometry(this);
+
+				for (var i = 0; i < preparedAnnotationRenderables.Count; i++)
```

Comment:
We are looping over `preparedAnnotationRenderables` the second time now. In theory we could remove the second loop and make the calls to `RenderDebugGeometry` in the first. On the other hand, I don't think that bit of performance is important when having the debug visualisations enabled, and as-is the code is imho cleaner.

Reply:
This is intentional - otherwise we risk rendering debug geometry behind annotation geometry.